diff --git a/datefudge.c b/datefudge.c
index de9619f..00cb012 100644
--- a/datefudge.c
+++ b/datefudge.c
@@ -62,8 +62,8 @@ time_t time(time_t *x) {
     return res;
 }
 
-int __gettimeofday(struct timeval *x, void *y) {
-    static int (*libc_gettimeofday)(struct timeval *, void *) = NULL;
+int __gettimeofday(struct timeval *x, struct timezone *y) {
+    static int (*libc_gettimeofday)(struct timeval *, struct timezone *) = NULL;
     int res;
 
     if(!libc_gettimeofday)
@@ -74,7 +74,7 @@ int __gettimeofday(struct timeval *x, void *y) {
     return 0;
 }
 
-int gettimeofday(struct timeval *x, void *y) {
+int gettimeofday(struct timeval *x, struct timezone *y) {
     return __gettimeofday(x,y);
 }
 
